controllers/clusterclaims.go

```diff
@@ -0,0 +1,145 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+
+	"github.com/go-logr/logr"
+	odfv1alpha1 "github.com/red-hat-storage/odf-operator/api/v1alpha1"
+	extensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1alpha1 "open-cluster-management.io/api/cluster/v1alpha1"
+)
+
+type ClusterClaimKey string
+
+const (
+	OdfVersion            ClusterClaimKey = "odf-version"
+	StorageSystemName     ClusterClaimKey = "storage-system-name"
+	StorageClusterName    ClusterClaimKey = "storage-cluster-name"
+	CephFsid              ClusterClaimKey = "ceph-fsid"
+	RookCephMonSecretName string          = "rook-ceph-mon"
+	FsidKey               string          = "fsid"
+)
+
+type ClusterClaimCreator struct {
+	Context context.Context
+	Logger  logr.Logger
+	Client  client.Client
+	Values  map[ClusterClaimKey]string
+}
+
+func (r *StorageSystemReconciler) ensureClusterClaims(instance *odfv1alpha1.StorageSystem, logger logr.Logger) error {
+	ctx := context.TODO()
+	crd := extensionsv1.CustomResourceDefinition{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: "clusterclaims.cluster.open-cluster-management.io"}, &crd)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("clusterclaims.cluster.open-cluster-management.io CRD not found. skipping creation of clusterclaims")
+			return nil
+		}
+		return err
+	}
+
+	creator := ClusterClaimCreator{
+		Logger:  logger,
+		Context: ctx,
+		Client:  r.Client,
+		Values:  make(map[ClusterClaimKey]string),
+	}
+
+	odfVersion, err := creator.getOdfVersion(instance.Namespace)
+	if len(odfVersion) == 0 || err != nil {
```

Comment:
is there any particular reason for  "" to length conversion?

Reply:
No reason, i just like to make sure that the string does not have any unwanted spaces.